### ISSUE TITLE
Wrong enumeration of GPU's on Windows 11 Pro

### ISSUE THREAD

**Issue by ITConundrum** [chaiNNer-org/chaiNNer#1899]
Information:
- Chainner version: 0.18.9
- OS: Windows 11 Pro
- New (2023 model) Intel i7 with integrated iris Xe GPU and also a dedicated NVIDIA GeForce RTX 3050 GPU

Description: 
- In settings > Python > [PyTorch  |  NCNN  |  ONNX] - Drop down selection list for GPU (for all configs) has wrong index assigned to GPU's.
- The "windows system" has the dedicated NVIDIA GPU at index 1, while the integrated GPU is 0.
- chaiNNer appears to have the wrong Index with each GPU name, so in Chainner, it has the NVIDIA listed at index 0 when it is actually at 1 as far as the OS is concerned. This causes all the workloads to actually execute on the integrated GPU and not the NVIDIA (verified while watching GPU workloads running upscales).
![image](https://github.com/chaiNNer-org/chaiNNer/assets/137963729/2165d99c-f36a-47b9-82a8-3ac5f60750b7)

![image](https://github.com/chaiNNer-org/chaiNNer/assets/137963729/82772624-8222-4adf-99a1-c1f5ba28df30)
![image](https://github.com/chaiNNer-org/chaiNNer/assets/137963729/ec4c4bc2-537f-44e6-8df1-0ac923416952)
![image](https://github.com/chaiNNer-org/chaiNNer/assets/137963729/dfa96892-8a21-479a-bcbf-a607436ea580)

**Comment by joeyballentine** [comment]
Can you check if this is still an issue in the nightly version? https://github.com/chaiNNer-org/chaiNNer-nightly

**Comment by joeyballentine** [comment]
Oh wait, i misunderstood. So for pytorch and onnx this shouldn't matter as those don't support your integrated GPU.

As for ncnn, we get the order of the GPU list from ncnn itself, so the fact that it's out of order means there's something wrong there. Perhaps the list just needs to be reversed? I'd need more people to check that

**Comment by michalk118** [comment]
Chainner version: 0.19.1
OS: Windows 11 Pro 22H2

0 titan rtx
1 rtx 3090

Same behavior here. If I select 0, it calculates on 1 and vice versa. At least on Pytorch it is like this. I haven't tried the others yet.

**Comment by joeyballentine** [comment]
That's the first I've heard of pytorch having that issue (backwards list). I'll be changing how that list gets made soon so maybe that'll fix it

**Comment by Yaiol** [comment]
Chainner version: 0.19.2
OS: Windows 11 Pro 22H2
GPU 0: Intel UHD
GPU 1: NVidia 2080

Don't know if this can help.
On Pytorch setup, only the NVidia is listed as GPU 0

![PyTorch GPU 1](https://github.com/chaiNNer-org/chaiNNer/assets/38218161/fcf65373-335d-45b3-979f-503a2cb1cc14)

Upscaling are done on correct GPU 1: NVidia 2080

![PyTorch GPU 2](https://github.com/chaiNNer-org/chaiNNer/assets/38218161/4149f793-e9ea-4c97-a87b-bfd61c1c0252)

So all is working correctly on my side

**Comment by joeyballentine** [comment]
Yeah as far as pytorch is concerned, only Nvidia GPUs exist. So that's expected.

As for the main issue, I'm hoping it'll be addressed when I finish #2104
